nselib/url.lua

```diff
@@ -173,12 +177,31 @@ function parse(url, default)
     return ""
   end)
   -- get params
-  url = string.gsub(url, "%;(.*)", function(p)
-    parsed.params = p
-    return ""
-  end)
+  -- Split up the query, if necessary
+  if(parsed.query) then
+    parsed.params = {}
+    local values = stdnse.strsplit('&', parsed.query)
+    for i, v in ipairs(values) do
+      local name, value = table.unpack(stdnse.strsplit('=', v))
+      parsed.params[name] = value
+    end
+  end
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
+    parsed.is_folder = true
+  else
+    parsed.is_folder = false
+    local split_str = stdnse.strsplit('%.', parsed.path)
+    if(split_str and #split_str > 1) then
+      parsed.extension = split_str[#split_str]
```

Comment:
This is where not parsing URI params will present a problem. We need to be able to detect the extension in the following URI: `http://example.com/index.php;PHPSESSIONID=blahblahblah`. It should be "php" in this case.